### PR TITLE
allow custom docker build

### PR DESCRIPTION
**Pull request by Stibbons**
to ease users behind a corporate proxy (that requires custom `docker build` parameters)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,8 @@ DOCKER_BUILD_ARGS := $(if $(DOCKER_VERSION), "--build-arg=DOCKER_VERSION=$(DOCKE
 DOCKER_RUN_OPTS := $(TRAEFIK_ENVS) $(TRAEFIK_MOUNT) "$(TRAEFIK_DEV_IMAGE)"
 DOCKER_RUN_TRAEFIK := docker run $(INTEGRATION_OPTS) -it $(DOCKER_RUN_OPTS)
 DOCKER_RUN_TRAEFIK_NOTTY := docker run $(INTEGRATION_OPTS) -i $(DOCKER_RUN_OPTS)
+DOCKER_BUILD := docker build
```
Comment (timoreimann) [inline Makefile]: I understood from the PR description that you want to customize the build parameters, but here you're customizing the build command.

I see how that could be used to customized both, but wouldn't it possibly be better to just specify the parameters?

Either way, wouldn't you also want to inject your customizations through, say, an environment variable, and read that in the Makefile? Right now, it looks like you'd have to modify your local copy of the Makefile to achieve what you want.

Comment (Stibbons) [inline Makefile]: Yes I need a bunch of args to inject proxy settings.

With this patch I can do 

    make image DOCKER_BUILD='docker build --build-arg...'

Comment (timoreimann) [inline Makefile]: My point is that you should only need to do 

`make image DOCKER_BUILD_ARGS='--build-arg...'`

Comment (ldez) [inline Makefile]: Please don't review before PR was splitted.

**Comment by ldez** [comment]
Any news on this PR?

**Comment by ldez** [comment]
@Stibbons Any news on this PR? (This PR doesn't built correctly for a long time)

**Comment by Stibbons** [comment]
Failure wasn't my fault

File: Makefile [hunk 2]
```diff
@@ -21,6 +21,7 @@ REPONAME := $(shell echo $(REPO) | tr '[:upper:]' '[:lower:]')
 TRAEFIK_IMAGE := $(if $(REPONAME),$(REPONAME),"containous/traefik")
 INTEGRATION_OPTS := $(if $(MAKE_DOCKER_HOST),-e "DOCKER_HOST=$(MAKE_DOCKER_HOST)", -v "/var/run/docker.sock:/var/run/docker.sock")
 
+DOCKER_VERSION := ''
```
Comment (ldez) [inline Makefile]: could you remove this line?
This is the root cause of the failure.

Comment (Stibbons) [inline Makefile]: alright

**Comment by Stibbons** [comment]
Are you sure the next line won't fail?
```
DOCKER_BUILD_ARGS := $(if $(DOCKER_VERSION), "--build-arg=DOCKER_VERSION=$(DOCKER_VERSION)",)
```
when `DOCKER_VERSION` is not defined?

**Comment by ldez** [comment]
it's defined in our CI.

File: Makefile (hunk 2 above)
Comment (ldez) [inline Makefile]: > One or more build-args [DOCKER_VERSION] were not consumed, failing build.

Comment (ldez) [inline Makefile]: It's related to your change in the `build-webui` make target.

**Comment by ldez** [comment]
`DOCKER_VERSION]` is define in our CI.

> One or more build-args [DOCKER_VERSION] were not consumed, failing build.

It's related to your changes in the `build-webui` make target.

**Comment by emilevauge** [comment]
@Stibbons your PR is failing on our CI. Do you want to continue to work on it or do you want to close it ?

**Comment by Stibbons** [comment]
I don't see why semaphore CI complains.
```
One or more build-args [DOCKER_VERSION] were not consumed, failing build
```
I don't have any `DOCKER_VERSION` in my patch

**Comment by ldez** [comment]
@Stibbons because `DOCKER_VERSION` is define in our CI.

In this case `DOCKER_BUILD_ARGS` contains `--build-arg=DOCKER_VERSION=1.12.6`.

```make
DOCKER_BUILD_ARGS := $(if $(DOCKER_VERSION), "--build-arg=DOCKER_VERSION=$(DOCKER_VERSION)",)
```

and your changes add `DOCKER_BUILD_ARGS` to `build-webui` target:

```make
build-webui:
	docker build $(DOCKER_BUILD_ARGS) -t traefik-webui -f webui/Dockerfile webui
```

**Comment by Stibbons** [comment]
yes I see but I don't see why it complains DOCKER_VERSION hasn't been consumed. It is, by this `DOCKER_BUILD_ARGS =...` at least

**Comment by emilevauge** [comment]
@Stibbons: @ldez took the time to explain in details where your PR breaks the CI but you don't want to admit that simple fact:

>Failure wasn't my fault

>I don't see why semaphore CI complains.

As we don't have time to convince you anymore, I'm sorry but I will close this PR.

**Comment by Stibbons** [comment]
I agree there is an issue but I am not enough confortable with Dockerfile to fix it completely :(

**Pull request closed without merging**